### PR TITLE
fix: [PL-28906]: Populating the listing scope of template list API to not list the remote templates when old_git_sync is true

### PR DESCRIPTION
**Pull request by sumanmabd**
### Summary

Adding the Listing scope to the templates list API so that it will not list the remote templates when old_git_sync is true.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/72-templates-library/pages/TemplatesPage/TemplatesPage.tsx [hunk 1]
```diff
@@ -92,7 +92,16 @@ export default function TemplatesPage(): React.ReactElement {
   } = useMutateAsGet(supportingTemplatesGitx ? useGetTemplateMetadataList : useGetTemplateList, {
     body: {
       filterType: 'Template',
-      ...(templateType && { templateEntityTypes: [templateType] })
+      ...(templateType && { templateEntityTypes: [templateType] }),
+      ...(!supportingTemplatesGitx
```
Comment (yc147) [inline src/modules/72-templates-library/pages/TemplatesPage/TemplatesPage.tsx:96]: We are already sending all this data in query params.. why is it needed in body?

Comment (yc147) [inline src/modules/72-templates-library/pages/TemplatesPage/TemplatesPage.tsx:96]: Revert this change. Not needed here

**Comment by sumanmabd** [comment]
retrigger jest

**Review by depanshu-dhiman — APPROVED** [review]

**Review by yc147 — APPROVED** [review]
